src/stream/src/executor/aggregation/distinct.rs

```diff
@@ -37,13 +38,15 @@ type DedupCache = ManagedLruCache<CompactedRow, Box<[i64]>>;
 struct ColumnDeduplicater<S: StateStore> {
     cache: DedupCache,
     _phantom: PhantomData<S>,
+    metrics: Arc<StreamingMetrics>,
 }
 
 impl<S: StateStore> ColumnDeduplicater<S> {
-    fn new(watermark_epoch: &Arc<AtomicU64>) -> Self {
+    fn new(watermark_epoch: &Arc<AtomicU64>, metrics: Arc<StreamingMetrics>) -> Self {
```

Comment:
May be better to also pass in `ActorContextRef` here, so that no need to pass `actor_id` when `dedup`ing and `flush`ing.

Reply:
Passing `ActorContextRef is unfriendly for unit test.

Reply:
`ActorContext::create(123)` doesn't work?